### ISSUE TITLE
JavaTemplate fails to generate an Enum field.

### ISSUE THREAD

**Issue by traceyyoshima** [openrewrite/rewrite#2331]
Issue discovered using JavaTemplate to create a public static final field of enum in ReplaceDuplicateStringLiterals:

```java
    @Test
    fun changeEnumField() = rewriteRun(
        { spec -> spec.recipe(replaceToStringWithLiteralRecipe)},
        java("""
            enum A {
                ONE, TWO, THREE;
                private static final String example = "VALUE".toString();
            }
        """,
            """
            enum A {
                ONE, TWO, THREE;
                private static final String example = "VALUE";
            }
        """)
    )
```

The problem is caused by `BlockStatementTemplateGenerator#enumClassDeclaration` starting at pos 1.
Note: there are a handful of JavaTemplate issues open that look to be related to changes made through 9/22.

**Comment by traceyyoshima** [comment]
Unassigning myself from the issue since Sam is looking into the JavaTemplate.

**Comment by pway99** [comment]
fixed by: #2446
